src/hooks/useMergeTransactions.ts

```diff
@@ -57,18 +59,29 @@ function useMergeTransactions({mergeTransaction}: UseMergeTransactionsProps): Us
     let [sourceTransactionReport] = useOnyx(`${ONYXKEYS.COLLECTION.REPORT}${getNonEmptyStringOnyxID(sourceTransaction?.reportID)}`, {
         canBeMissing: true,
     });
+    let [targetTransactionPolicy] = useOnyx(`${ONYXKEYS.COLLECTION.POLICY}${getNonEmptyStringOnyxID(targetTransactionReport?.policyID)}`, {
+        canBeMissing: true,
+    });
+    let [sourceTransactionPolicy] = useOnyx(`${ONYXKEYS.COLLECTION.POLICY}${getNonEmptyStringOnyxID(sourceTransactionReport?.policyID)}`, {
+        canBeMissing: true,
+    });
```

Comment:
@youssef-lr, coming from this comment https://github.com/Expensify/App/pull/73248/#discussion_r2636319481, are you happy with putting those codes here, or should we split it into another hook?